src/plugincollection.js

```diff
@@ -58,20 +58,27 @@ CKEDITOR.define( [ 'collection', 'promise', 'log' ], function( Collection, Promi
 					CKEDITOR.require( [ 'plugin!' + plugin ],
 						// Success callback.
 						function( LoadedPlugin ) {
-							var loadedPlugin = new LoadedPlugin( that._editor );
-							loadedPlugin.name = plugin;
-							loadedPlugin.path = CKEDITOR.getPluginPath( plugin );
-							loadedPlugin.deps = getPluginDeps( plugin );
+							var deps = getPluginDeps( plugin );
+							var isPluginDep = plugin.indexOf( '/' ) > 0;
```

Comment:
There should be no real difference if all your plugins and paths are correct, but it's an interesting question in case of some mistakes.

* Checking by path makes sure that you can only have a plugin file under a specific path.
* Checking by `instanceof` will allow you to have more plugins inside one plugin package what perhaps makes some sense, but then we would need to make sure that they really work there (if we want it to become a feature).
* Not checking `instanceof` is also unsafe because someone could mistakenly return a wrong value in his/her plugin module.

So I'm for keeping the current check and adding `instanceof` check inside. If `LoadedPlugin` is not an instance of `Plugin` then I would throw an error.